packages/expo/src/experimental/LocalCredentials.tsx

```diff
@@ -0,0 +1,150 @@
+import { useClerk, useSignIn } from '@clerk/clerk-react';
+import type { SignInResource } from '@clerk/types';
+import { AuthenticationType, isEnrolledAsync, supportedAuthenticationTypesAsync } from 'expo-local-authentication';
+import * as SecureStore from 'expo-secure-store';
+import type { PropsWithChildren } from 'react';
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+type LocalCredentials = {
+  identifier: string;
+  password: string;
+};
+
+type BiometricType = 'fingerprint' | 'face-recognition';
+
+export const LocalAuthContext = createContext<{
+  setCredentials: (creds: LocalCredentials) => Promise<void>;
+  hasCredentials: boolean;
+  clearCredentials: () => void;
+  authenticate: () => Promise<SignInResource>;
+  biometryType: BiometricType | null;
+}>({
+  // @ts-expect-error Initial value cannot return what the type expects
+  setCredentials: () => {},
+  hasCredentials: false,
+  clearCredentials: () => {},
+  // @ts-expect-error Initial value cannot return what the type expects
+  authenticate: () => {},
+  biometryType: null,
+});
+
+export const useLocalCredentials = () => {
+  return useContext(LocalAuthContext);
+};
+
+const useEnrolledBiometric = () => {
+  const [isEnrolled, setIsEnrolled] = useState(false);
+
+  useEffect(() => {
+    let ignore = false;
+
+    void isEnrolledAsync().then(res => {
+      if (ignore) {
+        return;
+      }
+      setIsEnrolled(res);
+    });
+
+    return () => {
+      ignore = true;
+    };
+  }, []);
+
+  return isEnrolled;
+};
+
+const useAuthenticationType = () => {
+  const [authenticationType, setAuthenticationType] = useState<BiometricType | null>(null);
+
+  useEffect(() => {
+    let ignore = false;
+
+    void supportedAuthenticationTypesAsync().then(numericTypes => {
+      if (ignore) {
+        return;
+      }
+      if (numericTypes.length === 0) {
+        return;
+      }
+
+      if (numericTypes.includes(AuthenticationType.FINGERPRINT)) {
+        setAuthenticationType('fingerprint');
+      } else {
+        setAuthenticationType('face-recognition');
+      }
+    });
+
+    return () => {
+      ignore = true;
+    };
+  }, []);
+
+  return authenticationType;
+};
+
+export function LocalCredentialsProvider(props: PropsWithChildren): JSX.Element {
+  const { isLoaded, signIn } = useSignIn();
+  const { publishableKey } = useClerk();
+  const key = `__clerk_local_auth_${publishableKey}_identifier`;
```

Comment:
💡Nit: Could we extract this interpolation to a separate function? Or perhaps type it, to avoid accidentally changing it or misspelling between individual variables.